sdk/aqueduct/decorator.py

```diff
@@ -368,6 +368,62 @@ def _convert_memory_string_to_mbs(memory_str: str) -> int:
     return multiplier * int(memory_scalar_str)
 
 
+def _update_operator_spec_with_engine_and_resource(
```

Comment:
Can we split this into two methods, one for updating `engine` and one for `resources`.